eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -188,35 +188,28 @@ object Helpers {
 
     for {
       script_opt <- extractShutdownScript(open.temporaryChannelId, localFeatures, remoteFeatures, open.upfrontShutdownScript_opt)
       willFund_opt <- LiquidityAds.validateRequest(nodeParams.privateKey, open.temporaryChannelId, fundingScript, open.fundingFeerate, isChannelCreation = true, open.requestFunding_opt, addFunding_opt.flatMap(_.rates_opt), open.useFeeCredit_opt)
     } yield (channelFeatures, script_opt, willFund_opt)
   }
 
-  private def validateChannelType(channelId: ByteVector32, channelType: SupportedChannelType, channelFlags: ChannelFlags, openChannelType_opt: Option[ChannelType], acceptChannelType_opt: Option[ChannelType], localFeatures: Features[InitFeature], remoteFeatures: Features[InitFeature]): Option[ChannelException] = {
-    acceptChannelType_opt match {
-      case Some(theirChannelType) if acceptChannelType_opt != openChannelType_opt =>
-        // if channel_type is set, and channel_type was set in open_channel, and they are not equal types: MUST reject the channel.
-        Some(InvalidChannelType(channelId, channelType, theirChannelType))
-      case None if Features.canUseFeature(localFeatures, remoteFeatures, Features.ChannelType) =>
-        // Bolt 2: if `option_channel_type` is negotiated: MUST set `channel_type`
-        Some(MissingChannelType(channelId))
-      case None if channelType != ChannelTypes.defaultFromFeatures(localFeatures, remoteFeatures, channelFlags.announceChannel) =>
-        // If we have overridden the default channel type, but they didn't support explicit channel type negotiation,
-        // we need to abort because they expect a different channel type than what we offered.
-        Some(InvalidChannelType(channelId, channelType, ChannelTypes.defaultFromFeatures(localFeatures, remoteFeatures, channelFlags.announceChannel)))
-      case _ =>
-        // we agree on channel type
-        None
+  private def validateChannelType(channelId: ByteVector32, channelType: SupportedChannelType, openChannelType_opt: Option[ChannelType], acceptChannelType_opt: Option[ChannelType]): Option[ChannelException] = {
+    if (openChannelType_opt.isEmpty || acceptChannelType_opt.isEmpty) {
+      Some(MissingChannelType(channelId))
+    } else if (!openChannelType_opt.contains(channelType) || !acceptChannelType_opt.contains(channelType)) {
+      Some(InvalidChannelType(channelId, acceptChannelType_opt.get))
+    } else {
+      // we agree on channel type
+      None
     }
   }
 
   /** Called by the funder of a single-funded channel. */
   def validateParamsSingleFundedFunder(nodeParams: NodeParams, channelType: SupportedChannelType, localFeatures: Features[InitFeature], remoteFeatures: Features[InitFeature], open: OpenChannel, accept: AcceptChannel): Either[ChannelException, (ChannelFeatures, Option[ByteVector])] = {
```

Comment:
Do we really need the `channelType` parameter ? it should be enough to check that the channel type defined in `accept` matches the one in `open` ?